api/logging/__init__.py

```diff
@@ -0,0 +1,32 @@
+from .logger import DefaultLogger
+
+VALID_LOG_LEVELS = {'debug', 'info', 'warning', 'error', 'critical'}
+
+def parse_log_entry(_logger, entry):
+    """
+    Parse a log entry expected from PCM frontend and logs
+    every single entry with the correct log level
+    returns
+        log level,
+        message,
+        extra dict (if present)
+    """
+    if 'level' not in entry or 'message' not in entry:
+        raise ValueError('Request body missing one or more mandatory fields ["message", "level"]')
+
+    level, message, extra = entry.get('level'), entry.get('message'), entry.get('extra')
+    if not (extra is None or type(extra) is dict):
+        raise ValueError('Extra param must be a valid json object')
+
+    lowercase_level = level.lower()
+    if lowercase_level not in VALID_LOG_LEVELS:
+        raise ValueError('Level param must be a valid log level')
+
+    return lowercase_level, message, extra
+
+def log_from_frontend(_logger, level, message, extra):
```

Comment:
question: why do we need to be as specific as implying a log is coming from the frontend in this case?

Reply:
the reason is that this function takes care of logging implementing a custom logic that is only valid for logs coming from the frontend (which right now is only doing `extra['source'] = 'frontend'` but may change in the future).
In the BE code we call the logger directly

Reply:
I will leave the considerations on the naming and purpose to other reviewers.

Reply:
After some thought I renamed the function to `push_log_entry` and it now accepts a new parameter (mandatory) called `source` to let the caller specify the source of the log, even though right now that value is always "frontend"